sdk/storage/AzureStorageBlob/Source/Transport/URLSessionTransferManager+TransferDelegate.swift

```diff
@@ -59,34 +62,42 @@ extension URLSessionTransferManager: TransferDelegate {
                     return
                 }
             } else {
-                delegate?.transfer(transfer, didUpdateWithState: state)
+                delegate?.transfer(transfer, didUpdateWithState: state, andProgress: nil)
             }
         }
-        saveContext()
+        if let context = (transfer as? NSManagedObject)?.managedObjectContext {
+            save(context: context)
+        }
     }
 
     func transfersDidUpdate(_ transfers: [Transfer]) {
-        let restorationIds = transfers.compactMap { ($0 as? TransferImpl)?.clientRestorationId }
-        for restorationId in Set(restorationIds) {
-            guard let delegate = client(forRestorationId: restorationId) as? TransferDelegate else { continue }
-            let transfersForId = transfers.filter { ($0 as? TransferImpl)?.clientRestorationId == restorationId }
-            delegate.transfersDidUpdate(transfersForId)
+        guard let restorationId = (transfers.first as? TransferImpl)?.clientRestorationId else { return }
```

Comment:
Is this a regression, or are we sure that all transfers will share the same restoration ID?

Reply:
I think probably a bad rebase. We don't want to assume all restoration IDs are the same.

Reply:
Fixed